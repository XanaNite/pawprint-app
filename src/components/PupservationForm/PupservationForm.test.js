import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PupservationForm from './PupservationForm';

describe(`PupservationForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders PupservationForm by default', () => {
    const wrapper = shallow(<PupservationForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the PupservationForm given props', () => {
    const wrapper = shallow(<PupservationForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
