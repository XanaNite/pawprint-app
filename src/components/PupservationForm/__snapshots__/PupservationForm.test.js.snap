// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PupservationForm component renders PupservationForm by default 1`] = `
<form
  className="pupservationForm"
  onSubmit={[Function]}
>
  <h3
    className="pupservation-header"
  >
    Pick a date:
  </h3>
  <div
    className="dateDropdown"
  >
    <t
      allowSameDay={false}
      dateFormat="MM/dd/yyyy"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      inlineFocusSelectedMonth={false}
      monthsShown={1}
      name="dateList"
      nextMonthButtonLabel="Next Month"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={2020-01-07T02:45:31.925Z}
      shouldCloseOnSelect={true}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={false}
      strictParsing={false}
      timeCaption="Time"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
    />
  </div>
  <h3
    className="pupservation-header"
  >
    Who's coming?
  </h3>
  <select
    className="listOfPups"
    name="pupList"
  />
  <h3
    className="pupservation-header"
  >
    Services:
  </h3>
  <select
    className="listOfServices"
    name="serviceList"
  >
    <option
      value="Grooming"
    >
      Grooming
    </option>
    <option
      value="Vet"
    >
      Vet
    </option>
    <option
      value="Daycare"
    >
      Daycare
    </option>
  </select>
  <section
    className="note"
  >
    <label>
      Note
    </label>
    <input
      name="note"
      type="text"
    />
  </section>
  <button
    className="pupservationConfirm-container"
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`PupservationForm component renders the PupservationForm given props 1`] = `
<form
  className="pupservationForm"
  onSubmit={[Function]}
>
  <h3
    className="pupservation-header"
  >
    Pick a date:
  </h3>
  <div
    className="dateDropdown"
  >
    <t
      allowSameDay={false}
      dateFormat="MM/dd/yyyy"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      inlineFocusSelectedMonth={false}
      monthsShown={1}
      name="dateList"
      nextMonthButtonLabel="Next Month"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={2020-01-07T02:45:31.956Z}
      shouldCloseOnSelect={true}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={false}
      strictParsing={false}
      timeCaption="Time"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
    />
  </div>
  <h3
    className="pupservation-header"
  >
    Who's coming?
  </h3>
  <select
    className="listOfPups"
    name="pupList"
  />
  <h3
    className="pupservation-header"
  >
    Services:
  </h3>
  <select
    className="listOfServices"
    name="serviceList"
  >
    <option
      value="Grooming"
    >
      Grooming
    </option>
    <option
      value="Vet"
    >
      Vet
    </option>
    <option
      value="Daycare"
    >
      Daycare
    </option>
  </select>
  <section
    className="note"
  >
    <label>
      Note
    </label>
    <input
      name="note"
      type="text"
    />
  </section>
  <button
    className="pupservationConfirm-container"
    type="submit"
  >
    Submit
  </button>
</form>
`;
